base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -429,7 +422,7 @@ where T: BlockchainBackend
         // This is a bit of a hack. The block hash *is* the header hash; so I just call header.hash here to avoid
         // having to reconstruct the block. BUT, things will break if block.hash() changes.
         let block_hash = header.hash();
-        self.update_metadata(block_height, block_hash, block_total_difficulty)?;
```

Comment:
wont this be needed be peers? Altho we might need to send over more than one no

Reply:
Not sure I understand the Q?

Reply:
Meta data is used by peers te determine if the need to sync if I am not mistaken. To determine syncing, they need the difficulty to know of their peer has a higher pow then them.

Reply:
You could be right. 

This refactor pulls total_difficulty out of the block header, and so right now there's nothing to share. There are a few other temp changes that are acknowledged as stop gaps to prevents blockers